camel/data_collector/alpaca_collector.py

```diff
@@ -0,0 +1,71 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from typing import Dict, List, Optional, Self, Union
+
+from camel.agents.chat_agent import ChatAgent
+from camel.data_collector.base import BaseDataCollector
+from camel.messages.base import BaseMessage
+
+
+class AlpacaDataCollector(BaseDataCollector):
+    def __init__(self):
+        super().__init__()
+        self.system_message: Optional[BaseMessage] = None
+        self.agent_name: Optional[str] = None
+
+    def inject(
+        self,
+        agent: Union[List[ChatAgent], ChatAgent],
+        name: Optional[Union[str, List[Optional[str]]]] = None,
+    ) -> Self:
+        r"""Inject an agent into the data collector.
+
+        Args:
+            agent (Union[List[ChatAgent], ChatAgent]):
+                The agent to inject.
+            name (Optional[Union[str, List[Optional[str]]]], optional):
+                The name of the agent. Defaults to None.
+        """
+        if len(self.agents) > 1:
+            raise ValueError("AlpacaDataCollector only supports one agent")
+        if isinstance(agent, list):
+            if len(agent) != 1:
+                raise ValueError("AlpacaDataCollector only supports one agent")
```

Comment:
based on current design I think we can further support list of agents and multiple msg, could we add this support with this PR?